kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusJobListener.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.services.prometheus;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.Summary;
+import org.jbpm.executor.AsynchronousJobEvent;
+import org.jbpm.executor.AsynchronousJobListener;
+import org.kie.api.executor.RequestInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.commons.lang3.StringUtils.defaultString;
+
+public class PrometheusJobListener implements AsynchronousJobListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PrometheusJobListener.class);
+
+    protected static final Gauge numberOfRunningJobs = Gauge.build()
+            .name("kie_server_job_running_total")
+            .help("Kie Server Running Jobs")
+            .labelNames("container_id", "command_name")
+            .register();
+
+    protected static final Counter numberOfJobsScheduled = Counter.build()
+            .name("kie_server_job_scheduled_total")
+            .help("Kie Server Started Jobs")
+            .labelNames("container_id", "command_name")
+            .register();
+
+    protected static final Counter numberOfJobsExecuted = Counter.build()
+            .name("kie_server_job_executed_total")
+            .help("Kie Server Executed Jobs")
+            .labelNames("container_id", "failed", "command_name")
+            .register();
+
+    protected static final Counter numberOfJobsCancelled = Counter.build()
+            .name("kie_server_job_cancelled_total")
+            .help("Kie Server Cancelled Jobs")
+            .labelNames("container_id", "command_name")
+            .register();
+
+    protected static final Summary jobDuration = Summary.build()
+            .name("kie_server_job_duration_seconds")
+            .help("Kie Server Job Duration")
+            .labelNames("container_id", "command_name")
+            .register();
+
+    private ThreadLocal<Summary.Timer> threadLocalStartTime = new ThreadLocal<>();
```

Comment:
@lucamolteni regarding threadlocal, this is a listener that is called before and after the job execution, I would say the we can use both approaches, a timestamp in the event or this one. For this one in particular is just that we dont have the time unit for the event, so less intrusive way to collect the data.

Reply:
Less intrusive but more fragile, ThreadLocal may give problems when used in ThreadPools

https://techblog.bozho.net/threadlocal-variables-and-thread-pools-it-can-go-wrong/

Reply:
I have also noticed that before it was static, now it isn't. Is there any reason for that?